persistence/sql/identity/persister_identity.go

```diff
@@ -297,6 +297,38 @@ func (p *IdentityPersister) createVerifiableAddresses(ctx context.Context, conn
 	return batch.Create(ctx, &batch.TracerConnection{Tracer: p.r.Tracer(ctx), Connection: conn}, work)
 }
 
+func (p *IdentityPersister) FindIdentityByAnyCredentialIdentifier(ctx context.Context, identifier string) (*identity.Identity, error) {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.FindIdentityByAnyCredentialIdentifier")
+	defer otelx.End(span, nil)
+
+	identifier = NormalizeIdentifier(identity.CredentialsTypePassword, identifier)
+
+	joins := `
+			INNER JOIN identity_credentials ic ON ic.identity_id = identities.id
+			INNER JOIN identity_credential_types ict ON ict.id = ic.identity_credential_type_id
+			INNER JOIN identity_credential_identifiers ici ON ici.identity_credential_id = ic.id`
+	wheres := `
+			AND (ic.nid = ? AND ici.nid = ? AND ici.identifier = ?)`
+
+	nid := p.NetworkID(ctx).String()
+	args := []any{nid, nid, identifier}
+	query := fmt.Sprintf(`
+		SELECT DISTINCT identities.*
+		FROM identities AS identities
+		%s
+		WHERE
+		%s
+		ORDER BY identities.id ASC
+		LIMIT 1`,
+		joins, wheres)
```

Comment:
Since you're not changing the `joins` and `wheres` variables after their declaration, I'd prefer seeing this all be one string. This way, it's hard to read what's going on.

And you can get rid of the `fmt.Sprintf` for the SQL statement, which can be potentially dangerous, if some one adjusts this code in the future (SQL injections!).